libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -815,21 +815,18 @@ def _insert_embeddings(
             raise
         return ids
 
-    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> Optional[bool]:
+    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> bool:
         """Delete embeddings in the vectorstore by the ids.
 
         Args:
-            ids: List of IDs to delete.
+            ids: List of IDs to delete. If None, delete all. Default is None.
+                No data is deleted if empty list is provided.
             kwargs: vectorstore specific parameters.
 
         Returns:
             Optional[bool]
         """
 
-        if ids is None or len(ids) == 0:
-            logging.info(EMPTY_IDS_ERROR_MESSAGE)
-            return False
-
         result = self._delete_texts_by_ids(ids)
         if result == 0:
             logging.info(INVALID_IDS_ERROR_MESSAGE)
```

Comment:
INVALID IDS ERROR MESSAGE should only be logged if len(ids) > 0 and result == 0.